docs/website/docs/general-usage/credentials/config_specs.md

```diff
@@ -0,0 +1,449 @@
+```
+#### Defaults
+
+If configuration is not provided, `dlt` uses the default credentials using `DefaultAzureCredential`.
+
+## Working with alternatives of credentials (Union types)
+
+If your source/resource allows for many authentication methods, you can support those seamlessly for
+your user. The user just passes the right credentials and `dlt` will inject the right type into your
+decorated function.
+
+Example:
+
+```python
+@dlt.source
+def zen_source(credentials: Union[ZenApiKeyCredentials, ZenEmailCredentials, str] = dlt.secrets.value, some_option: bool = False):
+  # depending on what the user provides in config, ZenApiKeyCredentials or ZenEmailCredentials will be injected in `credentials` argument
+  # both classes implement `auth` so you can always call it
+  credentials.auth()
+  return dlt.resource([credentials], name="credentials")
+
+# pass native value
+os.environ["CREDENTIALS"] = "email:mx:pwd"
+assert list(zen_source())[0].email == "mx"
+
+# pass explicit native value
+assert list(zen_source("secret:🔑:secret"))[0].api_secret == "secret"
+
+# pass explicit dict
+assert list(zen_source(credentials={"email": "emx", "password": "pass"}))[0].email == "emx"
+
+```
+
+> This applies not only to credentials but to all specs (see next chapter).
+
+Read the [whole test](https://github.com/dlt-hub/dlt/blob/devel/tests/common/configuration/test_spec_union.py), it shows how to create unions
+of credentials that derive from the common class, so you can handle it seamlessly in your code.
+
+## Writing custom specs
+
+**specs** let you take full control over the function arguments:
+
+- Which values should be injected, the types, default values.
+- You can specify optional and final fields.
+- Form hierarchical configurations (specs in specs).
+- Provide own handlers for `on_partial` (called before failing on missing config key) or `on_resolved`.
+- Provide own native value parsers.
+- Provide own default credentials logic.
+- Adds all Python dataclass goodies to it.
+- Adds all Python `dict` goodies to it (`specs` instances can be created from dicts and serialized
+  from dicts).
+
+This is used a lot in the `dlt` core and may become useful for complicated sources.
+
+In fact, for each decorated function a spec is synthesized. In case of `google_sheets` following
+class is created:
+
+```python
+from dlt.sources.config import configspec, with_config
+
+@configspec
+class GoogleSheetsConfiguration(BaseConfiguration):
+  tab_names: List[str] = None  # manadatory
+  credentials: GcpServiceAccountCredentials = None # mandatory secret
+  only_strings: Optional[bool] = False
+```
+
+- All specs derive from [BaseConfiguration.](https://github.com/dlt-hub/dlt/blob/devel/dlt/common/configuration/specs/base_configuration.py#L170)
+
+- All credentials derive from [CredentialsConfiguration.](https://github.com/dlt-hub/dlt/blob/devel/dlt/common/configuration/specs/base_configuration.py#L307)
+
+> Read the docstrings in the code above.
```

Comment:
This line does not make it clear what docstring should be read and why. Let's also tone down it and be more suggestive rather than imperative. 
It's best remove the blockquote as nothing is quoted.